tests/diff/collection_name_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// TestCollectionName documents difference in reposnes:
+//   * dots
+//   * dashes
+//   * max length in FerretDB: 255, in MongoDB: 63.
+//   * FerretDB reserved prefix is _ferretdb_, MongoDB - system..
+func TestCollectionName(t *testing.T) {
+	t.Parallel()
+
+	ctx, db := setup(t)
+	db = db.Client().Database("admin")
```

Comment:
There is no need to use `admin` database, the per-test database returned by `setup` will do just fine. That also removes the need to drop collections

Reply:
> There is no need to use `admin` database.

Applied, looks good.

> That also removes the need to drop collections

However, It doesn't work w/o deletion.

Reply:
`setup` creates unique empty database for each test. There are no collections to drop.

If it does not work, please demonstrate that

Reply:
> `setup` creates unique empty database for each test. There are no collections to drop.
> 
> If it does not work, please demonstrate that

Well, it was not defined clearly what you asked about, so I tried if it work with a database created for 2 tests, not for each separately. 

If you want to prove it, you can run dance tests against MongoDB with setup placed under `t.Run("ReservedPrefix", func(t *testing.T) {` not under `t.Run("FerretDB", func(t *testing.T) {`. 


Of course, when we initialize a database each time for each test, it works now.